internal/handler/users_handler.go

```diff
@@ -79,3 +80,99 @@ func GetUsersHandler(store *db.Store) gin.HandlerFunc {
 		ctx.JSON(http.StatusOK, user)
 	}
 }
+
+// ListUsers handler
+type listUsersRequest struct {
+	Limit  int32 `form:"limit" binding:"required,min=1"`
+	Offset int32 `form:"offset" binding:"required,min=0"`
+}
+
+func ListUsersHandler(store *db.Store) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req listUsersRequest
+		if err := ctx.ShouldBindQuery(&req); err != nil {
+			ctx.JSON(http.StatusBadRequest, errorResponse(err))
+			return
+		}
+
+		args := db.ListUsersParams{
+			Limit:  req.Limit,
+			Offset: req.Offset,
+		}
+
+		users, err := store.ListUsers(ctx, args)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+			return
+		}
+		ctx.JSON(http.StatusOK, users)
+	}
+}
+
+// Delete user//
+// type deleteUsersRequest struct {
+// 	ID int64 `uri:"id" binding:"required,min=1"`
+// }
+
+// func DeleteUsersHandler(store *db.Store) gin.HandlerFunc {
+// 	return func(ctx *gin.Context) {
+// 		var req deleteUsersRequest
+// 		if err := ctx.ShouldBindUri(&req); err != nil {
+// 			ctx.JSON(http.StatusBadRequest, errorResponse(err))
+// 			return
+// 		}
+
+// 		err := store.DeleteUsers(ctx, req.ID)
+// 		if err != nil {
+// 			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+// 			return
+// 		}
+// 		ctx.Status(http.StatusNoContent)
+// 	}
+// }
```

Comment:
Please do not leave in commented out code.

Reply:
Ready to check again.